src/features/Lessons/LessonDetailPage/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { Link, useParams, useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { Container, Pagination } from '@edx/paragon';
+
+import Table from 'features/Main/Table';
+import { columns } from 'features/Lessons/LessonDetailPage/columns';
+import { fetchStudentsDataSuccess, updateCurrentPage } from 'features/Students/data/slice';
+
+import { fetchStudentsData } from 'features/Students/data';
+
+import { initialPage } from 'features/constants';
+
+const LessonDetailPage = () => {
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { classId, lessonId } = useParams();
+
+  const institutionRef = useRef(undefined);
+  const [currentPage, setCurrentPage] = useState(initialPage);
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const students = useSelector((state) => state.students.table);
+
+  const handlePagination = (targetPage) => {
+    setCurrentPage(targetPage);
+    dispatch(updateCurrentPage(targetPage));
+  };
+
+  useEffect(() => {
+    const initialTitle = document.title;
+
+    document.title = lessonId;
+
+    return () => {
+      document.title = initialTitle;
+    };
+  }, [lessonId]);
+
+  useEffect(() => {
+    const initialState = {
+      results: [],
+      count: 0,
+      numPages: 0,
+    };
+
+    if (institution.id) {
+      const params = {
+        course_name: classId,
+        class_name: lessonId,
+        limit: true,
+      };
+      dispatch(fetchStudentsData(institution.id, initialPage, params));
+    }
+
+    return () => {
+      dispatch(fetchStudentsDataSuccess(initialState));
```

Comment:
Why is necessary call success action with initial state that is not the real data?

Reply:
At the moment to go back using the back arrow 
![f](https://github.com/Pearson-Advance/frontend-app-institution-portal/assets/52179095/6732eaa7-4a07-417f-b9f4-f2c8fc52e655)
Or browser arrow, and then select a new class this page will show you the previous results of your result, this could be confusing so picture this:

You use this path:
`http://localhost:1980/courses/my-test-course/class-1`
Then the page loads something like
![carbon](https://github.com/Pearson-Advance/frontend-app-institution-portal/assets/52179095/394a60ec-9f0b-4274-a705-551eb7afe6d4)

Now you go back to 
http://localhost:1980/courses/

And pick a new course and class
`http://localhost:1980/courses/**my-test-course-2**/class-1`

For a short moment the component will show the previous result while loading new information this is more visible when the network speed is slow.

So in order to avoid this scenario we just clean the info of that class.

Reply:
Understood, so In that case you should create a reducer to reset the data. We should follow the principle of single responsibility and in the case of reducers with request, success and failure have the purpose of indicating the status of a service call